ovos_core/skill_installer.py

```diff
@@ -0,0 +1,174 @@
+import sys
+from os.path import join, exists
+from subprocess import Popen, PIPE
+from tempfile import gettempdir
+from typing import Optional
+
+from combo_lock import ComboLock
+from ovos_config.config import Configuration
+
+from ovos_bus_client import Message
+from ovos_utils.log import LOG
+from importlib import reload
+import ovos_plugin_manager
+
+
+class SkillsStore:
+    # default constraints to use if none are given
+    DEFAULT_CONSTRAINTS = '/etc/mycroft/constraints.txt'
+    PIP_LOCK = ComboLock(join(gettempdir(), "ovos_pip.lock"))
+
+    def __init__(self, bus, config=None):
+        self.config = config or Configuration().get("skills", {}).get("installer", {})
+        self.bus = bus
+        self.bus.on("ovos.skills.install", self.handle_install_skill)
+        self.bus.on("ovos.skills.uninstall", self.handle_uninstall_skill)
+        self.bus.on("ovos.pip.install", self.handle_install_python)
+        self.bus.on("ovos.pip.uninstall", self.handle_uninstall_python)
+
+    def shutdown(self):
+        pass
+
+    def play_error_sound(self):
+        self.bus.emit(Message("mycroft.audio.play_sound",
+                              {"uri": "snd/error.mp3"}))
+
+    def play_success_sound(self):
+        self.bus.emit(Message("mycroft.audio.play_sound",
+                              {"uri": "snd/acknowledge.mp3"}))
```

Comment:
Will this work with HiveMind, or if users have specified a custom error/acknowledge sound?

Reply:
doesnt matter for hivemind since this is not supposed to happen there, its only on core devices so any generated message is not supposed to be forwarded via hivemind

but the sound would work yes, this is a bundled default sound in ovos audio so the uri will resolve

Reply:
oh i missed the custom user sounds, in a future PR this should not use the acknowledge sound, we will want dedicated sounds for this stuff, it is just a placeholder